src/com/mendix/widget/GoogleMaps/components/Map.ts

```diff
@@ -1,99 +1,144 @@
+                    }
+                })
+            );
         }
+        this.setState({ locations: nextProps.locations });
     }
 
     render() {
-        return DOM.div({ className: "mx-google-maps", ref: (c) => this.mapDiv = c });
-    }
-
-    componentDidMount() {
-        if (!this.state.isLoaded) {
-            this.loadGoogleScript();
-        }
-    }
-
-    componentWillUnmount() {
-        google.maps.event.clearListeners(this.map, "resize");
+        return DOM.div({ className: "widget-google-maps" },
+            createElement(Alert, { message: this.state.alertMessage }),
+            createElement(GoogleMap, this.getGoogleMapProps(),
+                this.state.locations
+                    ? this.state.locations.map((locationObject: Location, index) =>
+                        (
+                            this.createMaker(locationObject, index)
+                        ))
+                    : undefined
+            )
+        );
     }
 
-    private loadGoogleScript(callback?: Function) {
-        const script = document.createElement("script");
-        script.src = `https://maps.googleapis.com/maps/api/js?key=${this.props.apiKey}`;
-        script.onload = () => {
-            this.setState({ isLoaded: true });
+    private getGoogleMapProps(): GoogleMapProps {
+        return {
+            bootstrapURLKeys: { key: this.props.apiKey },
+            center: this.state.center,
+            defaultZoom: 7,
+            onGoogleApiLoaded: () => this.handleOnGoogleApiLoaded(),
+            resetBoundsOnResize: true,
+            yesIWantToUseGoogleMapApiInternals: true
         };
-        script.onerror = () => {
-            mx.ui.error("Could not load Google Maps script. Please check your internet connection");
-        };
-        document.body.appendChild(script);
     }
 
-    private loadMap() {
-        const mapConfig: google.maps.MapOptions = { zoom: 14 };
-        this.map = new google.maps.Map(this.mapDiv, mapConfig);
-        this.getLocation(this.props.address !== undefined
-            ? this.props.address
-            : this.defaultCenter,
-            this.createMarker,
-            this.centerToDefault
-        );
-        google.maps.event.addDomListener(window, "resize", () => {
-            const center = this.map.getCenter();
-            google.maps.event.trigger(this.map, "resize");
-            this.map.setCenter(center);
-        });
+    private handleOnGoogleApiLoaded() {
+        this.setState({ isLoaded: true });
+        this.centerMap();
+        this.plotAddresses();
     }
 
-    private getLocation(address: string, successCallback: Function, failureCallback?: Function) {
-        const geocoder = new google.maps.Geocoder();
-        geocoder.geocode({ address }, (results, status) => {
-            if (status === google.maps.GeocoderStatus.OK) {
-                successCallback(results[0].geometry.location);
+    private centerMap() {
```

Comment:
I already made sure that the center is a must added property from the modeler. check `defaultCenterAddress` property